nimbus-consensus/src/lib.rs

```diff
@@ -222,14 +222,12 @@ where
 	let maybe_key = available_keys.into_iter().find(|type_public_pair| {
 		// Have to convert to a typed NimbusId to pass to the runtime API. Maybe this is a clue
 		// That I should be passing Vec<u8> across the wasm boundary?
-		NimbusApi::can_author(
-			&*client.runtime_api(),
-			&at,
-			NimbusId::from_slice(&type_public_pair.1).expect("Provided keys should be valid"),
-			slot_number,
-			parent,
-		)
-		.expect("NimbusAPI should not return error")
+		if let Ok(nimbus_id) = NimbusId::from_slice(&type_public_pair.1) {
+			NimbusApi::can_author(&*client.runtime_api(), &at, nimbus_id, slot_number, parent)
+				.unwrap_or_default()
```

Comment:
`unwrap_or(false)` might be more clear here (and less relient on defined types), although it's technically the same thing